LoopStructural/datatypes/_bounding_box.py

```diff
@@ -93,7 +93,10 @@ def nelements(self, nelements):
         box_vol = self.volume
         ele_vol = box_vol / nelements
         # calculate the step vector of a regular cube
-        step_vector = np.zeros(3)
+        if self.dimensions == 2:
```

Comment:
Thanks for picking this up, I think maybe it could just be `step_vector = np.zeros(self.dimensions)`